### DISCUSSION TITLE
🤔 Does anybody know why we had to use Awaitility?

### DISCUSSION THREAD

**Discussion by rogervinas** [AdevintaSpain/ms-test--hello-twitch#2] [Q&A] [answered]
On the [living code session #2](https://youtu.be/UW-DkoRI1FQ) all went well and the tests were 🟢 but ...

... when we pushed our code to master some logging tests failed on the CI 😨 and the solution 💡 was to use [Awaitility](https://github.com/awaitility/awaitility)!

You can see one example here 👉 [LoggingTestCase.kt#L43](https://github.com/AdevintaSpain/ms-test--hello-twitch/blob/master/src/integrationTest/java/com/adevinta/mstesthellotwitch1/infrastructure/testcases/LoggingTestCase.kt#L43)

Does anybody know the root cause? 🤔 

**First to answer correctly will win an Adevinta T-SHIRT 👕!**

**Comment by vanphuong12a2** [comment, marked as answer]
It's because we use the AsyncAppender for logging (https://github.com/AdevintaSpain/ms-test--hello-twitch/blob/master/src/main/resources/logback-spring.xml#L9) so the status could be returned before the log is written to the console.

**Reply by rogervinas** [reply]
Awesome @vanphuong12a2! The T-SHIRT is yours! 👏
